File: dotfiles/.config/sway/help-bar.py
#!/usr/bin/env python3
import json, time, sys

print(json.dumps({"version": 1}))
print("[")
print("[]")
sys.stdout.flush()

while True:
    print("," + json.dumps([{
        "full_text": "term:Mod+Enter  run:Mod+d  split:Mod+h/v  full:Mod+f  kill:Mod+Shift+q  reload:Mod+Shift+r  exit:Mod+Shift+e  ",
        "color": "#888888"
    }]))
    sys.stdout.flush()
    time.sleep(60)

File: dotfiles/.config/sway/help-bar.c
#include <stdio.h>
#include <unistd.h>

int main() {
    printf("{\"version\":1}\n[\n[]\n");
    fflush(stdout);
    while (1) {
        printf(",["
            "{\"full_text\":\"term:Mod+Enter  run:Ctrl+Space  "
            "split:Mod+h  full:Mod+f  kill:Mod+Shift+q  "
            "reload:Mod+Shift+r  exit:Mod+Shift+e  "
            "IME:Mod+Space\","
            "\"color\":\"#888888\"}"
            "]\n");
        fflush(stdout);
        sleep(60);
    }
}
